x-pack/plugins/viz_editor/server/query/to_table.ts

```diff
@@ -127,6 +127,20 @@ function rawToTable(select: SelectOperation[], result: any) {
   });
 }
 
+function getTypeByOp(op: SelectOperation) {
+  switch (op.operation) {
+    case 'date_histogram':
+      return 'date';
+    case 'avg':
+    case 'count':
+    case 'sum':
+      return 'number';
+    case 'col':
+      // TODO how do we get this information?
+      return 'number';
+  }
```

Comment:
This function is too simple- we can and probably will run aggregations that support multiple types of documents. Count specifically works for any type. We can't even use the JSON types to decide, for example the IP address type looks like a string but is not treated that way. So I suspect that this function needs to load index patterns at some level, and I would push on your definition here to be more accurate.

Reply:
You are right, it sounds like index pattern + operation would be sufficient to get the type, right? If we pull this logic into the datasource-plugin on the client side (see other comment) and `tablify` there, this should work just fine.